### PR TITLE
Update Device logic nd functions

### PR DESCRIPTION
**Pull request by kvish**
Updated DeviceLogic ND functions

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kvish__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by kapil-ms** [comment]
Can you also update where the functions UpdateDeviceFromDeviceInfoPacketAsync and ExtractDevicePropertyValuesModels are called ?

**Comment by kvish** [comment]
These are called from EventHubs and MVC controllers so we want to hold off until the ND classes are done

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogicND.cs [hunk 1]
```diff
@@ -416,6 +458,40 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
             return registryRepositoryDevice;
         }
 
+        public async Task<DeviceND> UpdateDeviceEnabledStatusAsyncND(string deviceId, bool isEnabled)
+        {
+            DeviceND registryRepositoryDevice = null;
+            dynamic repositoryDevice = null;
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogicND.cs]: forgot dynamic ?

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogicND.cs [hunk 2]
```diff
@@ -416,6 +458,40 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
...
+            ExceptionDispatchInfo capturedException = null;
+
+            // if an exception happens at this point pass it up the stack to handle it
+            await _iotHubRepository.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled);
+
+            try
+            {
+                repositoryDevice = await _deviceRegistryCrudRepository.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled);
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogicND.cs]: isn't there a UpdateDeviceEnabledStatusAsyncND?

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogicND.cs [hunk 3]
```diff
@@ -416,6 +458,40 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
...
+                registryRepositoryDevice = TypeMapper.Get().map<DeviceND>(repositoryDevice);
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogicND.cs]: if we remove dynamic we don't have to do mapping.

**Pull request merged**